narwhals/_compliant/series.py

```diff
@@ -436,3 +439,76 @@ class EagerSeriesStructNamespace(  # type: ignore[misc]
     StructNamespace[EagerSeriesT_co],
     Protocol[EagerSeriesT_co, NativeSeriesT_co],
 ): ...
+
+
+class _EagerSeriesHist(Generic[NativeSeriesT]):
```

Comment:
This'll fix the `mypy` warnings like:

```py
narwhals/_compliant/series.py:469: error: Missing return statement  [empty-body]
        def is_empty_series(self) -> bool: ...
```

With `Generic`, you'd need to add [`@abc.abstractmethod`](https://docs.python.org/3/library/abc.html#abc.abstractmethod) to them, but a nice thing about [`Protocol`](https://mypy.readthedocs.io/en/stable/protocols.html#defining-subprotocols-and-subclassing-protocols) is that `...`

> In particular, omitting a value for an attribute or a method body will make it implicitly abstract

```suggestion
class _EagerSeriesHist(Protocol[NativeSeriesT]):
```

Reply:
@FBruzzesi all the warnings are now resolved (https://github.com/narwhals-dev/narwhals/pull/2882#commits-pushed-974369b) 🎉 

FWIW I don't actually like what I've wrote in those commits, but its a starting point for you to build on I guess 😅